fxa-auth-db-server/test/backend/remote.js

```diff
@@ -88,1354 +89,1319 @@ function captureFailureEvent(t, server) {
+          })
+          .then(function (r) {
+            respOk(r)
 
-          var result = r.obj
-          t.equal(result.length, 2, 'two emails returned')
-          t.equal(result[0].email, user.account.email, 'matches account email')
-          t.equal(!!result[0].isPrimary, true, 'isPrimary is true on account email')
-          t.equal(!!result[0].isVerified, !!user.account.emailVerified, 'matches account emailVerified')
+            var result = r.obj
+            assert.equal(result.length, 2, 'two emails returned')
+            assert.equal(result[1].email, secondEmailRecord.email, 'matches secondEmail email')
+            assert.equal(!!result[1].isPrimary, false, 'isPrimary is false on secondEmail email')
+            assert.equal(!!result[1].isVerified, true, 'matches secondEmail isVerified')
 
-          // Attempt to get a specific secondary email
-          return client.getThen('/email/' + emailToHex(thirdEmailRecord.email))
-        })
-        .then(function (r) {
-          respOk(t, r)
-
-          var result = r.obj
-          t.equal(result.email, thirdEmailRecord.email, 'matches email')
-          t.equal(!!result.isPrimary, false, 'isPrimary is false on email')
-          t.equal(!!result.isVerified, !!thirdEmailRecord.emailVerified, 'matches emailVerified')
-        })
-    }
-  )
-
-  test(
-    'add account, check password, retrieve it, delete it',
-    function (t) {
-      t.plan(35)
-      var user = fake.newUserDataHex()
-      return client.putThen('/account/' + user.accountId, user.account)
-        .then(function(r) {
-          respOkEmpty(t, r)
-          var randomPassword = Buffer(crypto.randomBytes(32)).toString('hex')
-          return client.postThen('/account/' + user.accountId + '/checkPassword', {'verifyHash': randomPassword})
-        })
-        .then(function(r) {
-          t.fail('should not be here, password isn\'t valid')
-        }, function(err) {
-          t.ok(err, 'incorrect password produces an error')
-          return client.postThen('/account/' + user.accountId + '/checkPassword', {'verifyHash': user.account.verifyHash})
-        })
-        .then(function(r) {
-          respOk(t, r)
-          var account = r.obj
-          t.equal(account.uid, user.accountId)
-          return client.getThen('/account/' + user.accountId)
-        })
-        .then(function(r) {
-          respOk(t, r)
-
-          var account = r.obj
-          var fields = 'accountId,email,emailCode,kA,verifierVersion,authSalt'.split(',')
-          fields.forEach(function(f) {
-            t.equal(user.account[f], account[f], 'Both Fields ' + f + ' are the same')
-          })
-          t.equal(user.account.emailVerified, !!account.emailVerified, 'Both fields emailVerified are the same')
-          t.notOk(account.verifyHash, 'verifyHash field should be absent')
-        }, function(err) {
-          t.fail('Error for some reason:' + err)
```

Comment:
I've gone through and made sure all `catch`s and `then(fail, check_err)` patterns are properly scoped.